woocommerce/woocommerce-ios
# HACK Week: Update media picker to support types other than images for downloadable products

Closes: #5064 


## What

This PR updates the fetching for WP media library to add an option to remove the image-only filter. This makes it possible to select document files from the media library for downloadable products.

## How
- Updated `MediaRemote` to add a new param `imagesOnly` to decide whether to apply the mime-type filter for the fetch requests.
- Updated `MediaAction` and `MediaStore` to add `imagesOnly` param.
- Renamed and updated `WordPressMediaLibraryPickerViewController` and `WordPressMediaLibraryPickerCoordinator` to add an option to not filter only images.
- Updated `ProductDownloadListViewController` to not apply the image filter when fetching data from the media library for downloadable products.
- Updated `MediaPickingCoordinator` to apply the image filter to work with images only.
- Updated `WordPressMediaLibraryPickerDataSource` to add `imagesOnly` param for filtering contents if needed.
- Updated `Media+WPMediaAsset` to support loading video files.

## Testing instructions

- Log in to a store and switch to the Products tab.
- Select and existing simple product or create a new one.
- In the product form, select the button to add a new photo.
- When the action sheet for photo source is displayed, select WP media library. Notice that only images are displayed in the list.
- Navigate back and select Downloadable Files (enable downloadable in ellipsis menu > Product Settings > Enable downloadable option if the option is not available).
- Select Add File > From WP Media library. Notice that all file types are displayed including document types, audios, videos, etc. Select any file, a new screen should be displayed to edit the file name and URL.

## Screenshots



https://github.com/woocommerce/woocommerce-ios/assets/5533851/f6b7b45d-b077-4fcc-922c-22c3e9311d81


---
